coreclient/src/conversations/persistence.rs

```diff
@@ -610,6 +641,68 @@ impl Conversation {
     }
 }
 
+pub async fn persist_message_status_report(
+    txn: &mut SqliteTransaction<'_>,
+    notifier: &mut StoreNotifier,
+    sender: &UserId,
+    report: &MessageStatusReport,
+) -> anyhow::Result<()> {
+    let (sender_uuid, sender_domain) = sender.clone().into_parts();
+
+    for update in &report.statuses {
+        let mimi_id = &update.mimi_id.to_vec();
+        let repr = update.status.repr();
+        let Some(message_id): Option<ConversationMessageId> = query_scalar!(
```

Comment:
I wonder if it possible to avoid this lookup for status changes with the same mimi id.

One way would be to create group mimi ids in a hashmap. Another maybe to combine the insert with a lookup. I tend to the first approach.